bfvmm/include/vmcs/vmcs_intel_x64.h

```diff
@@ -3453,187 +3380,794 @@ namespace guest_es_access_rights
-        constexpr const auto name = "type";
-
-        inline auto get()
-        { return (vm::read(addr, name) & mask) >> from; }
+        constexpr const auto name = "virtualize_apic_accesses";
 
-        template<class T> void set(T val)
-        { vm::write(addr, (vm::read(addr, name) & ~mask) | ((val << from) & mask), name); }
-    }
+        inline bool
+        is_enabled()
+        { return (vm::read(addr, name) & mask) != 0; }
 
-    namespace s
-    {
-        constexpr const auto mask = 0x0000000000000010UL;
-        constexpr const auto from = 4;
-        constexpr const auto name = "s";
+        template <class T> void set(T val)
+        { set_vm_control(val, msr_addr, addr, name, mask); }
 
-        inline auto get()
-        { return (vm::read(addr, name) & mask) >> from; }
+        template <class T> void set_if_allowed(T val, bool verbose) noexcept
+        { set_vm_control_if_allowed(val, msr_addr, addr, name, mask, verbose); }
 
-        template<class T> void set(T val)
-        { vm::write(addr, (vm::read(addr, name) & ~mask) | ((val << from) & mask), name); }
-    }
+        inline void enable()
+        { set(1UL); }
 
-    namespace dpl
-    {
-        constexpr const auto mask = 0x0000000000000060UL;
-        constexpr const auto from = 5;
-        constexpr const auto name = "dpl";
+        inline void disable()
+        { set(0UL); }
 
-        inline auto get()
-        { return (vm::read(addr, name) & mask) >> from; }
+        inline void enable_if_allowed(bool verbose = false) noexcept
+        { set_if_allowed(1UL, verbose); }
 
-        template<class T> void set(T val)
-        { vm::write(addr, (vm::read(addr, name) & ~mask) | ((val << from) & mask), name); }
+        inline void disable_if_allowed(bool verbose = false) noexcept
+        { set_if_allowed(0UL, verbose); }
     }
 
-    namespace present
+    namespace enable_ept
     {
-        constexpr const auto mask = 0x0000000000000080UL;
-        constexpr const auto from = 7;
-        constexpr const auto name = "present";
-
-        inline auto get()
-        { return (vm::read(addr, name) & mask) >> from; }
+        constexpr const auto mask = 0x0000000000000002UL;
+        constexpr const auto from = 1;
+        constexpr const auto name = "enable_ept";
 
-        template<class T> void set(T val)
-        { vm::write(addr, (vm::read(addr, name) & ~mask) | ((val << from) & mask), name); }
-    }
+        inline bool
+        is_enabled()
+        { return (vm::read(addr, name) & mask) != 0; }
 
-    namespace avl
-    {
-        constexpr const auto mask = 0x0000000000001000UL;
-        constexpr const auto from = 12;
-        constexpr const auto name = "avl";
+        template <class T> void set(T val)
+        { set_vm_control(val, msr_addr, addr, name, mask); }
 
-        inline auto get()
-        { return (vm::read(addr, name) & mask) >> from; }
+        template <class T> void set_if_allowed(T val, bool verbose) noexcept
```

Comment:
The diff is really hard to read, but did we still have a get? I cannot remember if we talked about that. We have a set, and set_if_allowed, we probably should have the same thing for get and get_if_allowed